src/app/charts/charts.component.ts

```diff
@@ -31,38 +34,59 @@ export class ChartsComponent implements OnInit {
     },
   ];
 
-  constructor() {}
+  constructor(private route: ActivatedRoute) {}
 
   ngOnInit(): void {
     this.lineChartData = this.labels.reduce(
       (acc, el) => [...acc, { data: [], label: el }],
       []
     );
-    this.loadChartData(this.labels[0]);
+
+    this.route.params.subscribe((params) =>
+      this.loadChartData(params["id"] || this.labels[0])
+    );
   }
 
-  onClick(type: string) {
+  onClick(type: any) {
     this.loadChartData(type);
   }
 
-  loadChartData(type: string) {
+  loadChartData(type: any) {
     this.activeSort = type;
+    let filteredData = [];
 
-    const fiteredData = this.chartData
-      .sort((a: ICountryInfo, b: ICountryInfo) => (a[type] < b[type] ? 1 : -1))
-      .slice(0, this.numberOfData);
+    if (this.period) {
+      filteredData = this.chartData.slice(0 - type);
+      this.text = `Covid-19 in ${this.chartData[0].Country}`;
+      if (typeof type === "string") {
+        this.activeSort = 0;
+      }
+    } else {
+      filteredData = this.chartData
+        .sort((a: ICountryInfo, b: ICountryInfo) =>
+          a[type] < b[type] ? 1 : -1
+        )
+        .slice(0, this.numberOfData);
+      this.text = `20 countries with most ${this.activeSort}`;
+    }
 
     this.lineChartData = this.lineChartData.reduce(
       (acc, el) => [
         ...acc,
         {
-          data: [...fiteredData.map((item) => item[el.label])],
+          data: [...filteredData.map((item) => item[el.label])],
           label: el.label,
         },
       ],
       []
     );
 
-    this.lineChartLabels = fiteredData.map((el) => el.Country);
+    this.lineChartLabels = filteredData.map((el) =>
+      this.period ? el.Date : el.Country
+    );
+  }
+
+  isClassActive(data) {
```

Comment:
This is simple enough to be in the template directly. Not a mistake, but keep it in mind. 🧄